src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -87,11 +81,55 @@ public struct MetricPoint
 
         public double DoubleValue { get; internal set; }
 
-        public long[] BucketCounts { get; internal set; }
+        internal MetricPointStatus MetricPointStatus { get; private set; }
 
-        public double[] ExplicitBounds { get; internal set; }
+        public long[] GetBucketCounts()
+        {
+            if (this.aggType == AggregationType.Histogram)
+            {
+                return this.histogramMeasurements.AggregatedBucketCounts;
+            }
+            else
+            {
+                throw new InvalidOperationException($"{nameof(this.GetBucketCounts)} is not supported for this metric type.");
+            }
+        }
 
-        internal MetricPointStatus MetricPointStatus { get; private set; }
+        public double[] GetExplicitBounds()
+        {
+            if (this.aggType == AggregationType.Histogram)
+            {
+                return this.histogramMeasurements.ExplicitBounds;
+            }
+            else
+            {
+                throw new InvalidOperationException($"{nameof(this.GetExplicitBounds)} is not supported for this metric type.");
+            }
+        }
+
+        public long GetHistogramCount()
+        {
+            if (this.aggType == AggregationType.Histogram || this.aggType == AggregationType.HistogramSumCount)
+            {
+                return this.histogramMeasurements.Count;
+            }
+            else
+            {
+                throw new InvalidOperationException($"{nameof(this.GetHistogramCount)} is not supported for this metric type.");
+            }
+        }
+
+        public double GetHistogramSum()
+        {
+            if (this.aggType == AggregationType.Histogram || this.aggType == AggregationType.HistogramSumCount)
+            {
+                return this.histogramMeasurements.Sum;
+            }
+            else
+            {
+                throw new InvalidOperationException($"{nameof(this.GetHistogramSum)} is not supported for this metric type.");
```

Comment:
Whatever we define (throw or just a log), I think it would be good to log the `aggType` or the metric type?

Reply:
`AggType` is `private`. We can create a mapping of `AggType` to `MetricType` and add that here. This could be done in a follow-up PR.